src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/UserChannel.cs

```diff
@@ -0,0 +1,149 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using System.Collections.Concurrent;
+using System.Reactive.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.Extensions.Logging;
+using MorganStanley.ComposeUI.Messaging;
+
+namespace DesktopAgent
+{
+    public class UserChannel
+    {
+        public string Name { get; }
+        private readonly ConcurrentDictionary<string, string> _contexts = new ConcurrentDictionary<string, string>();
+        private string? _lastContext = null;
+        private readonly string _topicRoot;
+        private readonly IMessageRouter _messageRouter;
+        private IAsyncObserver<TopicMessage> _broadcastObserver;
+        private ILogger<UserChannel> _logger;
+
+
+        public UserChannel(string name, IMessageRouter messageRouter, ILogger<UserChannel>? logger)
+        {
+            Name = name;
+            _topicRoot = $"composeui/fdc3/userchannels/{name}/";
+            _messageRouter = messageRouter;
+
+            _logger = logger;
+        }
+
+        public async ValueTask Connect()
+        {
+            await _messageRouter.ConnectAsync();
+
+            _broadcastObserver = AsyncObserver.Create<TopicMessage>(x => HandleBroadcast(x.Payload?.ToString()));
+
+            var broadcastSubscribing = _messageRouter.SubscribeAsync(_topicRoot + "broadcast", _broadcastObserver);
+            await broadcastSubscribing;
+
+            LogConnected();
+        }
+
+        internal ValueTask HandleBroadcast(string? payload)
```

Comment:
When would this method return something different than a `ValueTask.CompletedTask`?

Reply:
Whenever we actually add something that's async inside :) 
The only reason for this to return a ValueTask is to match the required signature for the subscription to the MessageRouter